### PR TITLE
feat(AEP-3): add AEP for glossary

### PR DESCRIPTION
**Pull request by toumorokoshi**
As a precursor to additional AEPs (e.g. resource types),
adding a glossary to ensure common terminology across
the AEP document.

### CODE REVIEW

File: aep/general/scope.yaml [hunk 1]
```diff
@@ -4,13 +4,15 @@ order: 0
...
+  - code: general
+    title: General
   - code: resources
     title: Resources
   - code: standard-methods
     title: Standard Methods
   - code: fields
     title: Fields
   - code: operations
-    title: Fields
```
Comment (rofrankel) [inline aep/general/scope.yaml:14]: Should this, and the other unrelated changes (e.g. `"` -> `'`) be in a separate PR?  Not a _huge_ deal if you don't want to separate them out, but it's nice to keep the blame history clean.

File: aep/general/0003/aep.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+Buffers by an `rpc` definition, or in HTTP via a `method` and a `path`.
+
+### API Request
+
+A single invocation of an API Method. It is often used as the unit for billing,
+logging, monitoring, and rate limiting.
+
+### API Resource
+
+An object that one or more API methods operate on.
```
Comment (rofrankel) [inline aep/general/0003/aep.md:61]: Nit:  this sentence ends with a preposition.  I am ambivalent about this, since avoiding doing so is a kind of archaic rule, but noting it nevertheless.

File: aep/general/0003/aep.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+### API Definition
+
+A well-structured representation of an API Service.
+
+### API Endpoint
+
+Refers to a network address that an API Service uses to handle incoming API
+Requests. One API Service may have multiple API Service Endpoints, such as
+`https://pubsub.googleapis.com` and `https://content-pubsub.googleapis.com`.
```
Comment (rofrankel) [inline aep/general/0003/aep.md:42]:
```suggestion
`https://pubsub.example.com` and `https://content-pubsub.example.com`.
```

Also, what about the `https://apis.example.com/pubsub` form?

File: aep/general/0003/aep.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  achieve desired state.
+- Ordering of these imperative actions.
+
+[Terraform][] is an example of such a client.
+
+### User
+
+A human being which is using an API directly, such as with cURL. This term is
+defined to differentiate usage in the AIPs between a human _user_ and a
+programmatic _client_.
```
Comment (rofrankel) [inline aep/general/0003/aep.md:101]: Should there be a term that's agnostic to human vs. program?

File: aep/general/0003/aep.md (hunk 3 above)
Comment (toumorokoshi) [reply aep/general/0003/aep.md:42]: de-googling I'm all for! I'm a bit mixed about the "adding a sub-api via the path", because it pollutes the resource name. I'm open to discussion though.

File: aep/general/0003/aep.md (hunk 2 above)
Comment (toumorokoshi) [reply aep/general/0003/aep.md:61]: I'm less opinionated about grammar, but fine with `An object upon which one or more API methods operate`. WDYT? Frankly my knowledge of english grammar is quite poor.

File: aep/general/0003/aep.md (hunk 4 above)
Comment (toumorokoshi) [reply aep/general/0003/aep.md:101]: I was thinking that too - what do you think is a good term? It's kind of a "client or user" relationship? 

or WDYT about "consumer"?

File: aep/general/scope.yaml (hunk 1 above)
Comment (toumorokoshi) [reply aep/general/scope.yaml:14]: I'm okay with minor unrelated changes like these, but if you're adamant I can separate!

File: aep/general/0003/aep.md (hunk 4 above)
Comment (toumorokoshi) [reply aep/general/0003/aep.md:101]: Added the term "consumer".

File: aep/general/0003/aep.md (hunk 2 above)
Comment (toumorokoshi) [reply aep/general/0003/aep.md:61]: Updated.

File: aep/general/scope.yaml (hunk 1 above)
Comment (rofrankel) [reply aep/general/scope.yaml:14]: Nah, I don't think this is the kind of "codebase" where it matters all that much.

File: aep/general/0003/aep.md (hunk 3 above)
Comment (rofrankel) [reply aep/general/0003/aep.md]: Well...what should an organization do if they don't have a domain dedicated to APIs?

For example, at Roblox, we have "apis.roblox.com".  We talked about getting a different domain or creating further subdomains (e.g. pubsub.apis.roblox.com) but both options got rejected (by people not organizationally local to the API platform).

File: aep/general/0003/aep.md (hunk 4 above)
Comment (rofrankel) [reply aep/general/0003/aep.md:95]: Yep, I think that's a good term, and it's what https://google.aip.dev/9 uses.

**Review by rofrankel — APPROVED** [review]

File: aep/general/0003/aep.md (hunk 3 above)
Comment (toumorokoshi) [reply aep/general/0003/aep.md]: sure, I think that's a good point. Here's an issue to discuss: https://github.com/aep-dev/aep.dev/issues/74.
